truenas/webui
# NAS-119698 / 22.12.1 / Unsupported type = cron and uri when trying to Launch a docker image (by AlexKarpov98)

Automatic cherry-pick failed. Please resolve conflicts by running:

    git reset --hard HEAD~1
    git cherry-pick -x 56603de217bbfaec55cfd6db971c698d98402964
    git cherry-pick -x 9b2d4a40a95d64b91c1cbb733b36a4aeddaae45b
    git cherry-pick -x 577b98f8479fb58a02f258919c0adc2d38c3d531
    git cherry-pick -x 6ac2b9d3f57e7b1579cd21513d269dbc5b8c15b9
    git cherry-pick -x 6858dda74a2a40bd037c8dd2c956e71c924a26db
    git cherry-pick -x 244c14b01529ed81de472532e28dfe598ac39a84
    git cherry-pick -x 00f974f095c84bf160921db341c45e912806529b
    git cherry-pick -x 28ba009f822884f15fc2e50f707418ebb9f12fc5
    git cherry-pick -x 142a271a738955ca91dbade4f0d60ff9040a9fe8
    git cherry-pick -x 2ddee8f14f0f1c340dce561365cf7b86d407aafa

Testing steps:
1. Make sure you are connected to `m40g3-137.dc1.ixsystems.net`
2. Go to APPS -> Available Applications -> Search Jacket (test_list_default)
3. Click Install 
<img width="393" alt="Screenshot 2023-01-18 at 15 48 24" src="https://user-images.githubusercontent.com/22980553/213188083-d75d7607-b156-45c1-8386-a1de3af2eb21.png">
4. At the top part of the form search for `CRON TEST` and `URI TEST`

What you will see is:
<img width="820" alt="Screenshot 2023-01-19 at 10 59 26" src="https://user-images.githubusercontent.com/22980553/213398937-a94b03d0-5fb7-43eb-914f-4352e5f3cf57.png">

Schema example:
```
        {
            "variable": "uri_test",
            "label": "URI TEST",
            "group": "General Settings",
            "schema": {
                "type": "uri"
            }
        },
        {
            "variable": "cron_test",
            "label": "CRON TEST",
            "group": "General Settings",
            "schema": {
                "type": "cron",
                "default": {
                    "hour": "*",
                    "minute": "*",
                    "month": "*",
                    "dom": "*",
                    "dow": "*"
                }
            }
        },
```


Original PR: https://github.com/truenas/webui/pull/7562
Jira URL: https://ixsystems.atlassian.net/browse/NAS-119698
